applications/GeoMechanicsApplication/tests/test_submerged_construction_of_excavation.py

```diff
@@ -30,102 +300,361 @@ def setUp(self):
+            expected_total_weight
+            + self.calculate_total_vertical_surface_load()
+            + self.calculate_weight_of_water_after_third_excavation()
+        )
+        self.check_vertical_reaction(
+            project_path,
+            self.stages_info["third_excavation"],
+            expected_total_vertical_reaction,
+        )
+
+    def create_wall_plots(self, project_path):
+        structural_stages = [
+            stage
+            for stage in self.stages_info.values()
+            if stage["base_name"] != "1_Initial_stage"
+            and stage["base_name"] != "2_Null_step"
+        ]
+        plot_titles = self.get_names_of_structural_stages(structural_stages)
+        y_coords_per_stage = self.get_y_coords_per_stage(
+            project_path, structural_stages
+        )
+
+        axial_force_plot_label = "Axial force"
+        axial_force_kratos_label = "AXIAL_FORCE"
+        axial_force_collections = self.get_variable_collections_per_stage(
+            axial_force_kratos_label,
+            axial_force_plot_label,
+            project_path,
+            structural_stages,
+            y_coords_per_stage,
+            extract_y_and_axial_force_from_line,
+        )
+
+        plot_utils.make_sub_plots(
+            axial_force_collections,
+            Path(project_path) / f"axial_forces_all_stages.svg",
+            titles=plot_titles,
+            xlabel="Axial Force [kN]",
+            ylabel="y [m]",
+        )
+
+        shear_force_plot_label = "Shear force"
+        shear_force_kratos_label = "SHEAR_FORCE"
+        shear_force_collections = self.get_variable_collections_per_stage(
+            shear_force_kratos_label,
+            shear_force_plot_label,
+            project_path,
+            structural_stages,
+            y_coords_per_stage,
+            extract_y_and_shear_force_from_line,
+        )
+        plot_utils.make_sub_plots(
+            shear_force_collections,
+            Path(project_path) / f"shear_forces_all_stages.svg",
+            titles=plot_titles,
+            xlabel="Shear Force [kN]",
+            ylabel="y [m]",
+        )
+
+        bending_moment_plot_label = "Bending moment"
+        bending_moment_kratos_label = "BENDING_MOMENT"
+        bending_moment_collections = self.get_variable_collections_per_stage(
+            bending_moment_kratos_label,
+            bending_moment_plot_label,
+            project_path,
+            structural_stages,
+            y_coords_per_stage,
+            extract_y_and_moment_from_line,
+        )
+        plot_utils.make_sub_plots(
+            bending_moment_collections,
+            Path(project_path) / f"bending_moments_all_stages.svg",
+            titles=plot_titles,
+            xlabel="Bending moment [kNm]",
+            ylabel="y [m]",
+        )
+
+    def get_variable_collections_per_stage(
+        self,
+        kratos_variable_label,
+        variable_plot_label,
+        project_path,
+        structural_stages,
```

Comment:
If I understand correctly, these are the base names of the stages. Perhaps the following suggested name makes that slightly clearer?
```suggestion
       stage_base_names,
```

Reply:
`structural_stages` is actually a list of the stage objects, from which we retrieve multiple things (i.e. the base name and the end times)

Reply:
Again, my bad ;-)